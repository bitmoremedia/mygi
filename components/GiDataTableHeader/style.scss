@import '../../css/variables';

.gi-data-table-header {
  display: none;
  padding-top: 15px;
  text-align: center;
  @include respond-to(tablet) {
    display: inline-block;
    width: 100%;
  }
}

h2.gi-data-table-header__title {
  margin-bottom: 0px;
}
