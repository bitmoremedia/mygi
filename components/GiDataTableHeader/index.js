import React, { PropTypes } from 'react'

import { giTypeDescr } from '../../utils'

import './style.scss'

const GiDataTableHeader = ({ categoryFilter, giTypeFilter }) => {
  let pageTitle = 'Glycemic Index'
  const giDescr = (giTypeFilter) ? giTypeDescr(giTypeFilter) : ''
  if (categoryFilter && !giTypeFilter) {
    pageTitle = `Glycemic Index of ${categoryFilter}`
  }
  if (!categoryFilter && giTypeFilter) {
    pageTitle = `${giDescr} GI Foods`
  }
  if (categoryFilter && giTypeFilter) {
    pageTitle = `${giDescr} GI ${categoryFilter}`
  }
  return (
    <div className="gi-data-table-header">
      <h2 className="gi-data-table-header__title">{pageTitle}</h2>
    </div>
  )
}

GiDataTableHeader.propTypes = {
  categoryFilter: PropTypes.string,
  giTypeFilter: PropTypes.oneOf(['', 'all', 'low-gi', 'medium-gi', 'high-gi']),
}

GiDataTableHeader.defaultProps = {
  categoryFilter: '',
  giTypeFilter: '',
}

export default GiDataTableHeader
