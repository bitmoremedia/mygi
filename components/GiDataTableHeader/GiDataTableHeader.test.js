import React from 'react'
import renderer from 'react-test-renderer'

import GiDataTableHeader from './index'

test('basic header with no props', () => {
  const component = renderer.create(
    <GiDataTableHeader />,
  )
  const tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
