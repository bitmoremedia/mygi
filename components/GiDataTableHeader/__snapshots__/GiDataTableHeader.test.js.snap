// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`basic header with no props 1`] = `
<div
  className="gi-data-table-header"
>
  <h2
    className="gi-data-table-header__title"
  >
    Glycemic Index
  </h2>
</div>
`;
